controlplane/pkg/local/endpoint_selector_service.go

```diff
@@ -92,9 +92,11 @@ func (cce *endpointSelectorService) Request(ctx context.Context, request *networ
 		// 7.1.1 Clone Connection to support iteration via EndPoints
 		newRequest, endpoint, err := cce.prepareRequest(ctx, request, clientConnection, ignoreEndpoints)
 		if err != nil {
+			span.Finish()
```

Comment:
'for' body can be extracted to a function to avoid hassle with span finishing

Reply:
I know, but not for this PR.